internal/langserver/handlers/complete.go

```diff
@@ -100,6 +100,11 @@ func CandidatesAtPos(data []byte, filename string, pos hcl.Pos, logger *log.Logg
 		return candidateList
 	}
 
+	// should not complete if the block is not resource
```

Comment:
This will disable the data source completion for msgraph provider. Please add a check here: https://github.com/Azure/ms-terraform-lsp/blob/main/internal/langserver/handlers/tfschema/azurerm_resource.go#L98

```go
return strings.HasPrefix(parts[1], "azurerm_") && parts[0] == "resource"
```

Then the completion and hover feature for data source will be disabled.